File: Chapter 03/Python/3.2 Coupling Exercise/After/user_report_generator.py
from user_data_service import UserDataService

class UserReportGenerator:
    def __init__(self):
        self.data_service = UserDataService()

    def generate_report(self, user_id):
        report = self.data_service.get_user_report(user_id)
        return f"User Report for {report.name} ({report.email}) - Total Spent: ${report.total_spent:.2f}"

File: Chapter 03/JS/3.2 Coupling Exercise/After/userReportGenerator.js
const UserDataService = require('./userDataService');

class UserReportGenerator {
    constructor() {
        this.dataService = new UserDataService();
    }

    generateReport(userId) {
        const report = this.dataService.getUserReport(userId);
        return `User Report for ${report.name} (${report.email}) - Total Spent: $${report.totalSpent.toFixed(2)}`;
    }
}

module.exports = UserReportGenerator;